File: android/app/src/main/java/edu/tjrac/swant/flutter/SingleFlutterEngineApp.java
package edu.tjrac.swant.flutter;

import android.app.Application;

import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.FlutterEngineCache;

import static edu.tjrac.swant.flutter.FlutterChannelWapper.FLAG_DEFAULT_FLUTTER_ENGINE;

public class SingleFlutterEngineApp extends Application {

    FlutterEngine flutterEngine;

    @Override
    public void onCreate() {
        super.onCreate();
        flutterEngine = new FlutterEngine(this);
        FlutterEngineCache.getInstance()
                .put(FLAG_DEFAULT_FLUTTER_ENGINE, flutterEngine);
    }

    @Override
    public void onTerminate() {
        flutterEngine.destroy();
        super.onTerminate();

    }
}

File: android/app/src/main/java/edu/tjrac/swant/flutter/SingleFlutterEngineApp.kt
package edu.tjrac.swant.flutter

import android.Manifest
import android.app.Application
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.FlutterEngineCache

class SingleFlutterEngineApp : Application() {
    var flutterEngine: FlutterEngine? = null
    override fun onCreate() {
        super.onCreate()
        val perms = arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE,Manifest.permission.WRITE_EXTERNAL_STORAGE)

        for(p in perms){
        }
        flutterEngine = FlutterEngine(this)
        FlutterEngineCache.getInstance()
            .put(FlutterChannelWapper.FLAG_DEFAULT_FLUTTER_ENGINE, flutterEngine)
    }

    override fun onTerminate() {
        flutterEngine?.destroy()
        super.onTerminate()
    }
}